Port learning repo's syntax examples to this repo

JordanMartinez opened the pull request:
Fixes #51.

While adding CI would be nice, this is at least better than what we have now.

thomashoneyman commented:
Thanks for adding this, @JordanMartinez -- I'm sorry for such a late response, I didn't realize I never followed up.

I'm glad to add examples to this repository, and I agree that adding CI would be nice. What do you think of moving these examples into a tests file, which would also give the benefit of a compiler-checked assertion of the output of these functions?

We could then trim the README to have a couple syntax examples with a longer explanation of how they work + point users to the tests for a larger set of examples (that are tested to work, unlike READMEs which tend to fall out of date).

JordanMartinez commented:
How's this?
- I've moved the code to `test/Examples.purs`
- Point readers to it via a relative link on the ReadMe
- Added a somewhat more complicated example of one parsing a small amount of CSV with some special handling

I don't think the module should be `Test.Examples` because purescript/package-sets#2 may one day get implemented.
